subsys/fs/Kconfig.fatfs

```diff
@@ -211,6 +212,11 @@ config FS_FATFS_WINDOW_ALIGNMENT
 	  that, in worst scenario, value provided here may cause FATFS
 	  structure to have size of twice the value.
 
+config FS_FATFS_REENTRANT
+	bool "FatFs reentrant"
+	help
+	  Enable the re-entrancy (thread safe) option of the FatFs
+
```

Comment:
Add more info on costs: for example that there will be mutex objects, one per volume. This is important because this takes some RAM, so people need to know.